test/TestMain.cpp

```diff
@@ -1,5 +1,5 @@
 #include <iostream>
-#include "PlayMain.h"
```

Comment:
Isn't the file `TestMain.cpp` supposed to be the unit test to test functions in `PlayMain.h`?

Reply:
As `PlayMain.h` is right now, the include wasn't needed, but if you are going to use functions declared in the header it is necessary. I just tired adding it back, and it seems like the changes made to the `CMakeLists.txt` files have fixed the issue. I would recommend using `#include <PlayMain.h>` instead of `#include "PlayMain.h"` as brackets are usually used for headers in additional include directories (passed to the compiler with `-I`) and quotes for paths relative to the compilation unit (the .cpp file being compiled), see [this question on StackOverflow](https://stackoverflow.com/questions/3162030/difference-between-angle-bracket-and-double-quotes-while-including-heade).

I can make a commit adding the include back if you want.

Reply:
Ok, that would be great. Thanks!

Reply:
It was the duplicate lines discussed above that caused the issue, as the target created by them doesn't specify `${PLAY}` in `target_link_libraries`.

The `${PLAY}` target includes its own directory containing `PlayMain.h` as `PUBLIC` in `target_include_directories` (`src/play/CMakeLists.txt` line 14). PUBLIC tells CMake that other targets depending on it should use that include directory as well. The `unit_tests` target specified in `test/CMakteLists.txt` does depend on `${PLAY}`, and as such uses the include directory and the compiler is able to find the header file. The executable target `tests` specified by the lines i removed does not depend on it, thus causing compilation to fail.

Reply:
Oh, why did duplication remove specification of `${PLAY}`? Weird. But ok, I think this is good now.

Reply:
It didn't remove the `${PLAY}` target. Both targets were there, and during a build of all targets, one of them failed, causing the entire build to fail.

Reply:
Hmm ok.